### PR TITLE
Added builder pattern for creating User instances

### PR DESCRIPTION
**Pull request by NimishaGarg**

### CODE REVIEW

File: src/main/java/controller/CreateUser.java [hunk 1]
```diff
@@ -49,7 +49,7 @@
             throw new UsernameAlreadyExistsException(path);
         } 
 
-        User newUser = new User(username);
+        User newUser = new User.Builder().username(username).build();
```
Comment (sankalpbansal-google) [inline src/main/java/controller/CreateUser.java:52]: Please check if we have User.newBuilder(). instead?

Comment (sankalpbansal-google) [reply src/main/java/controller/CreateUser.java:52]: here and elsewhere

Comment (NimishaGarg) [reply src/main/java/controller/CreateUser.java:52]: I have made a static class named Builder inside the User class, which I am using for creating User instances. I did not understand what you mean by `newBuilder()`.

Comment (AgamAgarwal) [reply src/main/java/controller/CreateUser.java:52]: You should make the constructor of `Builder` private and instead expose a public static method in `User` called `newBuilder` which calls the constructor and returns a new instance.

Comment (NimishaGarg) [reply src/main/java/controller/CreateUser.java]: Got it, Done.

File: src/main/java/entity/User.java [hunk 2]
```diff
@@ -5,11 +5,70 @@
...
+
+    public Builder picture(String picture) {
+      this.picture = picture;
+      return this;
+    }
+
+    public User build() {
+      User user = new User();
+      user.CreationTS = this.creationTs;
+      user.UserID = this.userId;
```
Comment (AgamAgarwal) [inline src/main/java/entity/User.java:62]: So the default is 0?

Comment (NimishaGarg) [reply src/main/java/entity/User.java:62]: Yes, we will set UserID later in that case.

**Pull request merged**